contrib/scripts/unloaddb.sh

```diff
@@ -0,0 +1,402 @@
+         WHEN "data_type" = 'FLOAT' THEN 4.0 \
+         WHEN "data_type" = 'DOUBLE' THEN 8.0 \
+         WHEN "data_type" = 'MONETARY' THEN 12.0 \
+         WHEN "data_type" = 'STRING' THEN prec \
+         WHEN "data_type" = 'VARCHAR' THEN prec \
+         WHEN "data_type" = 'NVARCHAR' THEN prec \
+         WHEN "data_type" = 'CHAR' THEN prec \
+         WHEN "data_type" = 'NCHAR' THEN prec \
+         WHEN "data_type" = 'TIMESTAMP' THEN 8.0 \
+         WHEN "data_type" = 'DATE' THEN 4.0 \
+         WHEN "data_type" = 'TIME' THEN 4.0 \
+         WHEN "data_type" = 'DATETIME' THEN 4.0 \
+         WHEN "data_type" = 'BIT' THEN FLOOR(prec / 8.0) \
+         WHEN "data_type" = 'BIT VARYING' THEN FLOOR(prec / 8.0) \
+         ELSE 0 \
+     END) as BIGINT)  AS [size] \
+ from db_attribute where class_name = '$table_name';"
+
+        if [ $# -eq 0 ];then
+                echo "0"
+                return
+        fi
+
+        size=$(csql $user $pass -l -c "$query" $db | grep "^<0000" | awk '{print $3}')
+        echo $size
+}
+
+function find_slot ()
+{
+        local selected=0
+        local i
+        local size=${slot_size[0]}
+
+        for ((i = 0; i < $num_proc; i++))
+        do
+                if [ ${slot_size[i]} -lt $size ];then
+                        size=${slot_size[i]}
+                        selected=$i
+                fi
+        done
+
+         echo $selected
+}
+
+function do_unloaddb ()
+{
+        local slot_num=$1
+        local i
+        local file="Tables_unloaded_$database.$slot_num"
+        local num_tables_in_slot=0
+        local msg="Success"
+        local pid
+
+        for ((i = 0; i < $num_tables; i++))
+        do
+                if [ ${slot[i]} -eq $slot_num ];then
+                        let "num_tables_in_slot++"
+                        echo "${table_selected[i]}" >> $file
+                fi
+        done
+
+        if [ $verbose = "yes" ];then
+                echo "Proc $slot_num: num tables: $num_tables_in_slot, ${slot_size[$slot_num]} bytes"
+        fi
+
+        echo "process $slot_num: starting" > unloaddb_$slot_num.status
+
+	cubrid unloaddb $user $pass --input-class-only --input-class-file $file $database &
+        pid=$!
+
+        echo $pid > unloaddb_$slot_num.pid
+
+        wait $pid
+
+        if [ $? -ne 0 ];then
+                msg="Failed"
+                echo "process $slot_num: failed" > unloaddb_$slot_num.status
+        else
+                echo "process $slot_num: success" > unloaddb_$slot_num.status
+        fi
+
+        if [ $verbose = "yes" ];then
+                echo "Finished Proc $slot_num ($msg)"
+        fi
+}
+
+function analyze_table_info ()
+{
+        local found=0
+        local line
+        local idx=0
+        local table_name
+        local db=$database
+
+        # Get all table names from CATALOG
+
+        if [ X$from_file != X"" ];then  # Read table name from file
+                result=$(cat $from_file)
```

Comment:
When tables is gotten from file, I seem like that you are better to set a flag and check the flag at below routines for readability.

Reply:
Comment accepted, I add a flag '**from_file**' for that and changed variable for file to 'filename'